src/Databases/MySQL/MySQLBinlogClientCatalog.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <Databases/MySQL/MySQLBinlogClient.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/Operators.h>
+
+namespace DB::MySQLReplication
+{
+
+class BinlogClientCatalog final : boost::noncopyable
```

Comment:
```suggestion
class BinlogClientFactory final : boost::noncopyable
```
Seems ok? Too many different names:)

Reply:
It does not only create/instantiate the binlog client, it also keeps track of them and returns if it was already existed.